src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -229,11 +231,46 @@ class PeerActor(
     context become waitingForConnectionResult(newConnection, noRetries)
   }
 
+  def handleBlockFastDownload(rlpxConnection: RLPxConnection): Receive = {
+    case RLPxConnectionHandler.MessageReceived(request: GetReceipts) =>
+      val receipts = request.blockHashes.slice(0, maxReceiptsPerMessage)
+        .flatMap(hash => storage.receiptStorage.get(hash))
+
+      rlpxConnection.sendMessage(Receipts(receipts))
+
+    case RLPxConnectionHandler.MessageReceived(request: GetBlockBodies) =>
+      val blockBodies = request.hashes.slice(0, maxBlocksBodiesPerMessage)
+        .flatMap(hash => storage.blockBodiesStorage.get(hash))
+
+      rlpxConnection.sendMessage(BlockBodies(blockBodies))
+
+    case RLPxConnectionHandler.MessageReceived(request: GetBlockHeaders) =>
+      val blockNumber = request.block.fold(a => Some(a), b => storage.blockHeadersStorage.get(b).map(_.number))
+
+      blockNumber match {
+        case Some(startBlockNumber) if startBlockNumber >= 0 && request.maxHeaders >= 0 && request.skip >= 0 =>
+
+          val headersCount: BigInt = if (maxBlocksHeadersPerMessage < request.maxHeaders) maxBlocksHeadersPerMessage else request.maxHeaders
+
+          val range = if (request.reverse) {
+            startBlockNumber to (startBlockNumber - (request.skip + 1) * headersCount + 1) by -(request.skip + 1)
+          } else {
+            startBlockNumber to (startBlockNumber + (request.skip + 1) * headersCount) by (request.skip + 1)
```

Comment:
While running the test `return block headers by block number` and looking at the value of `range` I noticed it was set to `NumericRange(3,4,5)` while the `headersCount` is set to 2, couldn't there be any problem with this calculation?